presto-delta/src/main/java/com/facebook/presto/delta/DeltaMetadata.java

```diff
@@ -150,8 +151,32 @@ public DeltaTableHandle getTableHandle(ConnectorSession session, SchemaTableName
             Optional<Set<ColumnHandle>> desiredColumns)
     {
         DeltaTableHandle tableHandle = (DeltaTableHandle) table;
-        ConnectorTableLayout layout = new ConnectorTableLayout(new DeltaTableLayoutHandle(tableHandle));
-        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+        if (!config.isFilterPushdownEnabled()) {
+            ConnectorTableLayout layout = new ConnectorTableLayout(new DeltaTableLayoutHandle(tableHandle));
+            return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+        }
+
+        // Split the predicate into partition column predicate and other column predicates
+        // Only the partition column predicate is fully enforced. Other predicate is partially enforced (best effort).
+        List<TupleDomain<ColumnHandle>> predicate = splitPredicate(constraint.getSummary());
+        TupleDomain<ColumnHandle> enforcedPredicate = predicate.get(0);
```

Comment:
This is unused. This is the partition predicate. Should this be the one used to create the "newDeltaTableHandle" since this is used to call iterateWithPredicate() which does partition pruning.

Reply:
New delta table handle needs both enforced and unenforced predicates. Enforced predicate is used in partition pruning, unenforced predicate is used in data skipping such as Parquet rowgroup pruning.